src/h2o_discovery/client.py

```diff
@@ -16,3 +22,71 @@ def get_services_uri(uri: str) -> str:
 
 def get_clients_uri(uri: str) -> str:
     return urllib.parse.urljoin(uri, CLIENTS_ENDPOINT)
+
+
+class Client:
+    """Synchronous Implementation of the Discovery Service API.
+
+    Listing methods do pagination and always return all of the available objects.
+    """
+
+    def __init__(self, uri: str):
+        self._uri = uri
+
+        self._environment_uri = get_environment_uri(uri)
+        self._services_uri = get_services_uri(uri)
+        self._clients_uri = get_clients_uri(uri)
+
+    def get_environment(self) -> model.Environment:
+        """Returns the information about the environment."""
+        with httpx.Client() as client:
```

Comment:
QQ: I'm wondering, shouldn't there be a way to pass extra arguments to the HTTP client constructor somehow? E.g. due to some customer headers or so?

Reply:
Yes, definitely. Timeout(s) at min. 

Materialized it as issue: https://github.com/h2oai/cloud-discovery/issues/141